core/Twig.php

```diff
@@ -65,7 +65,9 @@ function piwik_fix_lbrace($string)
 
 function piwik_escape_filter(Twig_Environment $env, $string, $strategy = 'html', $charset = null, $autoescape = false) {
 
-    $string = twig_escape_filter($env, $string, $strategy, $charset, $autoescape);
+    if ($strategy != 'whole_url') {
+        $string = twig_escape_filter($env, $string, $strategy, $charset, $autoescape);
```

Comment:
any reason this is not done for whole_url? wouldn't it be the same as simple adding like a filter `whole_url` that does below logic?

Reply:
When twig sees an escape like `|e('html_attr')` it doesn't apply the default `|e('html')` escape to a variable. But when twig sees `|e('whole_url')` it applies the default escape. So if we call `twig_escape_filter` above, it double encodes the URL. Only way I could come up w/ to not do that, is to make sure whole_url doesn't escape anything.

Reply:
My message was bit confusing, not sure if understood what I meant. Just to be sure... what I meant was we could add `whole_url` as a regular twig filter like this maybe:

```
    private function addFilter_wholeUrl()
    {
        $filter = new \Twig_SimpleFilter('wholeUrl', function ($value) {
            if (!UrlHelper::isLookLikeSafeUrl($string)) { ... } return $value
         });
        $this->twig->addFilter($filter);
    }
```

and then use it like `{{ link|wholeUrl|e('html_attr')}}`.
Then we wouldn't need the above `if`... as `whole_url` is not really escaping but more like filtering/changing the value? And using `|e('whole_url')` could give a developer the impression the value will be escaped there (and think another escape html_attr is not needed) which in reality it doesn't...

Reply:
Oh I see, yes, makes sense.